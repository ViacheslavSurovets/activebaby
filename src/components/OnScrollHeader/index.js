export { default as OnScrollHeader } from './OnScrollHeader';
