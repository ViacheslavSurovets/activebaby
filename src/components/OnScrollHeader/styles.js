import styled, { css } from 'styled-components';
import { desktop } from '@utils/media';

 const styleOnScrollHeaderWrapperVisible = css`
    visibility: visible;
    opacity: 1;
    width: 100%;
    height: inherit;
    color: inherit;
    transition: opacity 1s, visibility 1s;
    nav {
      position: absolute;
    }
`;

const styleOnScrollHeaderWrapperUnvisible = css`
  height: 0;
  width: 0;
  opacity: 0;
  visibility: hidden;
`;

export const getStylesScrollWrapper = ( props ) => {
  if ( props.visible ){
    return styleOnScrollHeaderWrapperVisible;
  } else {
    return styleOnScrollHeaderWrapperUnvisible;
  }
};

export const OnScrollHeaderWrapper = styled.div`
   ${desktop}{
      z-index: 10;
      position: sticky;
      top:0;
      background-color:#fff;
      
      ${getStylesScrollWrapper};
   }
`;
