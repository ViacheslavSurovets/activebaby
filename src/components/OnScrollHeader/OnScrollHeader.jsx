import React from 'react';
import { HeaderNavContainer } from '@components/HeaderNavContainer';
import { OnScrollHeaderWrapper } from './styles';
import PropTypes from 'prop-types';

const OnScrollHeader = ( { visible } ) => (
  <OnScrollHeaderWrapper visible={visible}>
    <HeaderNavContainer />
  </OnScrollHeaderWrapper>
);

export default OnScrollHeader;

OnScrollHeader.propTypes = {
  visible: PropTypes.bool.isRequired
};
