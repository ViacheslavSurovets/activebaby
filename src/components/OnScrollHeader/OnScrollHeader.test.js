import React from 'react';
import { shallow } from 'enzyme';
import OnScrollHeader from './OnScrollHeader';


describe ( 'OnScrollHeader', () => {

  it ( 'OnScrollHeader Snapshot', () => {
    expect ( shallow ( <OnScrollHeader visible /> ) ).toMatchSnapshot ();
  } );
} );
