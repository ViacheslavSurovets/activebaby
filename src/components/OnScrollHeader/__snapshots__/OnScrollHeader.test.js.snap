// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnScrollHeader OnScrollHeader Snapshot 1`] = `ShallowWrapper {}`;
